Components/CommentForm.jsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from "react";
+import { postArticleComment } from "./api";
+import UserContext from "./UserContext"
+import { useContext } from "react"
+import Error from "./Error";
+import Loading from "./Loading";
+
+
+export default function CommentForm({id, setCommentCount, addComment, removeComment}){
+      const {loggedInUser} = useContext(UserContext)
+       const { username } = loggedInUser;
+       const [postComment, setPostComment] = useState("")
+       const [commentError, setCommentError] = useState(false)
+       const [error, setError] = useState(null)
+       const [submitting, setSubmitting] = useState(false)
+       const [isLoading, setIsLoading] = useState(false)
+
+
+        const handleSubmit = (event) => {
+          setIsLoading(true)
+          event.preventDefault();
+          if (postComment === ""){
+            setCommentError(true)
+          } else {
+            setSubmitting(true)
+            setCommentCount((commentCount) => commentCount + 1)
+            const newComment = {
+            article_id: id,
+            author: username,
+            body: postComment,
+            comment_id: 10000,
```

Comment:
This could be dangerous, if you had a comment in the database with the ID of 10000 you may be able to accidentally delete it temporarily while you are waiting for the response with the real ID

Reply:
With comments it's preferable to wait until the conclusion of the response to add it to the comments array

Reply:
Your UI should still show some sort of indication the comment is posting, disabling the button etc

Reply:
If you are going to do it this way I would recommend adding an id that can never exist, -1 or "temp" something